### PR TITLE
Fix case where hom-ref "variant" with no data had wrong sized PLs and

### PR DESCRIPTION
**Pull request by ldgauthier**
didn't merge with adjacent block

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFUnitTest.java [hunk 1]
```diff
@@ -110,6 +110,12 @@ public void testLowQualVariantToGQ0HomRef() {
         Assert.assertTrue(!modified.filtersWereApplied());
         Assert.assertEquals(modified.getLog10PError(), VariantContext.NO_LOG10_PERROR);
 
+        final Genotype longPls = VariantContextTestUtils.makeG("sample1", Allele.NO_CALL, Allele.NO_CALL, 0,0,0,0,0,0);
+        final VariantContext lotsOfZeroPls = makeNoDepthVC("lowQualVar", Arrays.asList(LONG_REF, DELETION, Allele.NON_REF_ALLELE), LONG_REF.length(), longPls);
+        final VariantContext properlySubset = reblocker.lowQualVariantToGQ0HomRef(lotsOfZeroPls).make();
+        Assert.assertEquals(properlySubset.getGenotype(0).getPL().length, 3);
```
Comment (vruano) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFUnitTest.java:116]: Can you check the content of PL also is correnct? is it all 0s or '.'?

Comment (vruano) [reply src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFUnitTest.java:116]: Assert.assertEquals(...getPL(), new int [] { 0, 0, 0}) may just work.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java [hunk 2]
```diff
@@ -507,9 +507,14 @@ protected GenotypeBuilder changeCallToHomRefVersusNonRef(final VariantContext lo
...
+                //we care about the best alt even though it's getting removed because NON_REF should get the best likelihoods
+                //it shouldn't matter that we're passing in different alt alleles since the GenotypesContext only knows
+                // the called alleles and this is a reference genotype that will stay hom-ref
                 final GenotypesContext context = AlleleSubsettingUtils.subsetAlleles(lowQualVariant.getGenotypes(),
                         genotype.getPloidy(), lowQualVariant.getAlleles(), Arrays.asList(inputRefAllele, bestAlt),
-                        null, GenotypeAssignmentMethod.BEST_MATCH_TO_ORIGINAL, lowQualVariant.getAttributeAsInt(VCFConstants.DEPTH_KEY, 0), false);  //BEST_MATCH to avoid no-calling low qual genotypes
+                        null, GenotypeAssignmentMethod.BEST_MATCH_TO_ORIGINAL, //BEST_MATCH to avoid no-calling low qual genotypes
+                        lowQualVariant.getAttributeAsInt(VCFConstants.DEPTH_KEY, 0),
+                        true);  //emitEmptyPLs = true to make sure we always subset
```
Comment (vruano) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java:517]: Not sure what a true here results into looking at the subsetAllele code. If it is what you need and that behavior from subsetAlleles is stable then I don't a problem with that.

**Review by vruano — CHANGES_REQUESTED** [review]

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFUnitTest.java (hunk 1 above)
Comment (ldgauthier) [reply src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFUnitTest.java:116]: It does work, thanks.  This is done.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java (hunk 2 above)
Comment (ldgauthier) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java]: The alternative is to drop the PLs, which I do not want.  I've changed the parameter name to be a little more clear.

**Review by vruano — APPROVED** [review]
Once tests pass you can merge.

**Pull request merged**